### ISSUE TITLE
Avoid use of install script for /cvmfs and use pip instead

### ISSUE THREAD

**Issue by egede** [ganga-devs/ganga#1238]
We should retire the old install_script that is used for installation of the ganga version on /cvmfs. This will ensure that we have the identical libraries used everywhere and also will break our last dependence on /afs.

The best will be to implement this will be to install ganga inside a virtualenv on /cvmfs. However, we should provide a wrapper that means that the use of the virtualenv is not visible. I think something as simple as

```
#!/bin/sh
. <installdir>/bin/activate
ganga
```

should work. Would be a good idea to implement before #1229 .

**Comment by milliams** [comment]
When a script is installed into a virtualenv with pip it should automatically rewrite the shebang line to point to the python executable in said virtual environment which means that simply running the ganga script in `.../venv/bin/ganga/` should automatically use the correct Python interpreter.

**Comment by egede** [comment]
@milliams So it does, and the `sys.path` is just correct then. That makes life easier. Slightly tricky thing that we have to make sure that the virtualenv is created with the correct python version (the box that we install into /cvmfs on probably just have a bog standard python 2.6).

**Comment by milliams** [comment]
Indeed. In Python 3, you can ensure this by writing:

```console
$ python3 -m venv ganga_venv
```

since that explicitly specifies the python interepreter. But in Python 2, since virtualenv isn't bundled with Python, you have to do:

```console
$ virtualenv --python=/path/to/python2.7 ganga_venv
```

**Comment by egede** [comment]
@milliams To ge the `--python` tag to work is a bit of a hassle as it ends up with errors of the type 

```
[uegede@lxplus026]~% virtualenv -p `which python` ganga
Running virtualenv with interpreter /cvmfs/lhcb.cern.ch/lib/lcg/releases/LCG_88/Python/2.7.13/x86_64-centos7-gcc62-opt/bin/python
New python executable in /afs/cern.ch/user/u/uegede/ganga/bin/python
Installing setuptools, pip, wheel...
  Complete output from command /afs/cern.ch/user/u/uegede/ganga/bin/python - setuptools pip wheel:
  Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
  File "/cvmfs/lhcb.cern.ch/lib/lcg/releases/LCG_88/Python/2.7.13/x86_64-centos7-gcc62-opt/lib/python2.7/tempfile.py", line 32, in <module>
    import io as _io
  File "/cvmfs/lhcb.cern.ch/lib/lcg/releases/LCG_88/Python/2.7.13/x86_64-centos7-gcc62-opt/lib/python2.7/io.py", line 51, in <module>
    import _io
ImportError: /afs/cern.ch/user/u/uegede/ganga/lib64/python2.7/lib-dynload/_io.so: undefined symbol: _PyErr_ReplaceException
```
However, if `PYTHONHOME` is set correctly to point to the alternative python installation, then it works.

It is not obvious which python version to point to though. For LHCb, it would be obvious to use the one from the LHCb installation area, but if somebody else is using it and doesn't mount the lhcb part of /cvmfs, this will fail.

**Comment by egede** [comment]
OP, testing a bit more around, the following is the minimal thing that I can get working. Where is the script that we currently run on the /cvmfs installation machine?

```
export PYTHONHOME=/cvmfs/lhcb.cern.ch/lib/lcg/releases/LCG_88/Python/2.7.13/x86_64-slc6-gcc62-opt                                                                                                 
export LD_LIBRARY_PATH=$PYTHONHOME/lib
virtualenv -p $PYTHONHOME/bin/python ganga
unset PYTHONHOME
source ganga/bin/activate
pip install ganga
deactivate
```

**Comment by milliams** [comment]
That's annoying that you have to jump through those hoops but if it works, it works. It's worth noting that there's a small shortcut you can take so that instead of writing:
```
source ganga/bin/activate
pip install ganga
deactivate
```
you can do it one go with:
```
ganga/bin/pip install ganga
```
by calling the explicit path to the copy of pip inside the virtualenv.

**Comment by mesmith75** [comment]
Did we ever decide on a plan with virtualenv? I am quite keen to get it implemented and bring us up to date.

**Comment by mesmith75** [comment]
So this seems to be working ok.There is a version at `/cvmfs/ganga.cern.ch/Ganga/install/ipythonTest` that works for me. Let's see if there are any comments.

**Comment by mesmith75** [comment]
Fixed by #1296 #1229 #1298 #1299
